packages/hashi/src/mainClient.js

```diff
@@ -1,8 +1,11 @@
+import { ContentNodeResource } from 'kolibri.resources';
+import router from 'kolibri.coreVue.router';
```

Comment:
The other general thing is that aside from the dependency injection, the `mainClient` doesn't really need to know or care about most of the stuff we are doing here - so we can actually put most of this code in the `kolibri.js` file - because it is being instantiated in the iframeClient and the mainClient, we can keep the code together for easy cross reference.